clients/cli/src/unwrap.rs

```diff
@@ -0,0 +1,231 @@
+    crate::{
+        common::{
+            get_account_owner, get_mint, parse_pubkey, parse_token_program, process_transaction,
+        },
+        config::Config,
+        output::{format_output, println_display},
+        CommandResult,
+    },
+    clap::{value_parser, Args},
+    serde_derive::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_clap_v3_utils::{
+        input_parsers::signer::{SignerSource, SignerSourceParserBuilder},
+        keypair::signer_from_source,
+    },
+    solana_cli_output::{display::writeln_name_value, QuietDisplay, VerboseDisplay},
+    solana_hash::Hash,
+    solana_pubkey::Pubkey,
+    solana_remote_wallet::remote_wallet::RemoteWalletManager,
+    solana_signature::Signature,
+    solana_signer::Signer,
+    solana_transaction::Transaction,
+    spl_token_wrap::{get_wrapped_mint_address, get_wrapped_mint_authority, instruction::unwrap},
+    std::{
+        fmt::{Display, Formatter},
+        rc::Rc,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone, Debug, Args)]
+pub struct UnwrapArgs {
+    /// The address of the wrapped token account to unwrap from
+    #[clap(value_parser = parse_pubkey)]
+    pub wrapped_token_account: Pubkey,
+
+    /// The address of the escrow account holding the unwrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub escrow_account: Pubkey,
+
+    /// The address of the token account to receive the unwrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_token_recipient: Pubkey,
+
+    /// The amount of tokens to unwrap
+    #[clap(value_parser)]
+    pub amount: u64,
+
+    /// Signer source of transfer authority (to burn wrapped tokens)
+    /// if different from fee payer
+    #[clap(
+        long,
+        value_parser = SignerSourceParserBuilder::default().allow_all().build()
+    )]
+    pub transfer_authority: Option<SignerSource>,
+
+    /// The address of the unwrapped mint, queried if not provided
+    #[clap(long, value_parser = parse_pubkey)]
+    pub unwrapped_mint: Option<Pubkey>,
+
+    /// The address of the token program for the wrapped mint,
+    /// queried if not provided.
+    #[clap(long, value_parser = parse_token_program)]
+    pub wrapped_token_program: Option<Pubkey>,
+
+    /// The address of the token program for the unwrapped mint,
+    /// queried if not provided.
+    #[clap(long, value_parser = parse_token_program)]
+    pub unwrapped_token_program: Option<Pubkey>,
+
+    #[clap(long, value_parser = value_parser!(Hash))]
+    pub blockhash: Option<Hash>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UnwrapOutput {
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_mint_address: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_token_program: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub recipient_token_account: Pubkey,
+
+    pub amount: u64,
+
+    pub signatures: Vec<Signature>,
+}
```

Comment:
Wrap gives quite a bit more -- could this also include the escrow, wrapped mint address, and wrapped token account?